apps/frontend/src/pages/quiz/libs/components/analyzing/libs/helpers/create-animated-spans/create-animated-spans.helper.ts

```diff
@@ -0,0 +1,27 @@
+import React from "react";
+
+import { TextAnimationDelay } from "../../enums/enums.js";
+import styles from "./styles.module.css";
+
+const createAnimatedSpans = (
+	text: string,
+	baseDelay: number = TextAnimationDelay.BASE_DELAY,
+): React.ReactNode[] => {
+	const words = text.split(" ");
+
+	return words.map((word, index) =>
+		React.createElement(
```

Comment:
What if we use a custom hook? Can we avoid `React.createElement` in that case?

Reply:
No, instead of helper, create a hook `useAnimatedSpans`

Reply:
so when I convert it to a custom hook should the file be a ts or tsx? I can't find a way to return an element in a .ts file without using `React.createElement`.

Reply:
Okay, let's make it .tsx